File: linked-list/palindromelinklist.cpp
class Solution{
  public:
    
    //Function to check whether the list is palindrome.
    bool isPalindrome(Node *head)
    {
        stack<int> s;
 
    // push all elements of the linked list into the stack
    Node* node = head;
    while (node)
    {
        s.push(node->data);
        node = node->next;
    }
 
    // traverse the linked list again
    node = head;
    while (node)
    {
        // pop the top element from the stack
        int top = s.top();
        s.pop();
 
        // compare the popped element with the current node's data
        // return false if mismatch happens
        if (top != node->data) {
            return 0;
        }
 
        // advance to the next node
        node = node->next;
    }
 
    // we reach here only when the linked list is a palindrome
    return 1;
}
   };

File: linked-list/palindromelinklist.js
var isPalindrome = function(head) {
         let stack=[];
         let curr =head; 
         while(curr){      //traverse and inserting the elements into stack.
            stack.push(curr.val);
            curr=curr.next;
         }
         while(head){          //after inserting we will traverse again 
                               //by comparing head with popped value 
            if(head.val != stack.pop()){
                return false; //if the head value is not equal to the popped value
                              //then it is not a palindrome. and break it .
                break;
            }
            head=head.next; // and we will also change the position of head with traversing. 
         }
         return true;
};
